### PR TITLE
Pkg 237v3

### PR DESCRIPTION
**Pull request by M-Waszkiewicz-Anaconda**
update to 1.82.0 
[upstream](https://github.com/boostorg/boost/tree/boost-1.82.0)
[conda forge boost-cpp](https://github.com/conda-forge/boost-cpp-feedstock) - similar to our libboost subpackage
[conda forge boost](https://github.com/conda-forge/boost-feedstock) - similar to py-boost subpackage
- heavy refactor as code has high technical debt, no longer needed workarounds and patches
- some feature removed ( eg. XStudio ) as suggested by historical commit description
- removed unmaintained not documented logic, taken [KISS principle](https://en.wikipedia.org/wiki/KISS_principle)
- better tests including linker for libboost borrowed from Conda Forge

### CODE REVIEW

**Review by varlackc — COMMENTED** [review]
There are already a few PRs for this package that probably will have to be closed out. 

    For example PR # 12: 

    https://github.com/AnacondaRecipes/boost-feedstock/pull/12

**Review by varlackc — COMMENTED** [review]
There seem to be some old PRs that are still open, not sure what we should do with them.

  `Update to 1.74.0`
  https://github.com/AnacondaRecipes/boost-feedstock/pull/10

  `recipe: forward-port recipe to 1.81`
  https://github.com/AnacondaRecipes/boost-feedstock/pull/11

File: recipe/meta.yaml
Comment (skupr-anaconda) [inline recipe/meta.yaml:5]:
```suggestion
```
Looks like it's not an issue anymore

**Review by skupr-anaconda — APPROVED** [review]
LGTM but I would wait for 2 additional reviews from experienced folks

File: recipe/meta.yaml
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml]: yes, removed

**Review by varlackc — APPROVED** [review]
LGTM

File: abs.yaml
Comment (rafaelmartins) [inline abs.yaml:5]:
```suggestion
upload_without_merge: true

```

**Review by rafaelmartins — APPROVED** [review]

**Pull request merged**
